### ISSUE TITLE
Rampage falling off after 30 seconds regardless of duration.

### ISSUE THREAD

**Issue by Yuskia** [FeenixServerProject/Archangel_2.4.3_Bugtracker#3817]
The way rampage is supposed to work is that after each crit it gains a stack and refreshes the duration, up to a maximum of 5 stacks. Once at 5 stacks it should then have 30 seconds of the buff, at which point it will no longer auto refresh, but you can cast the buff again to refresh the duration at 5 stacks. However, the way it currently works, is that it will always remove itself 30 seconds after the initial cast, regardless of the duration of the buff.

**Comment by veremi** [comment]
Its Video from Exalibur private server, soo bad informant 
Here is Video from global https://www.youtube.com/watch?v=Hn8eMD8zdOo and as u can see, he refresh Rampage every 25-29 sec. But problem on Feenix is, if u refresh ur rampage x5, u lose all ur stacks and u have to stack it again.
Regards

**Comment by Yuskia** [comment]
You're right, my mistake. However the issue is still two fold, one that it cannot last longer than 30 seconds from the very first cast, and 2 that refreshing causes you to lose all stacks.

**Comment by veremi** [comment]
Its should be 30sec+ time for stackx5 from crit, time is depend on ur crit chance and rng is not static value .
Regards Wytoczka

**Comment by Yuskia** [comment]
You're clearly not understanding. The way the skill works, you cast it once, it has an initial duration of 30 seconds. Everytime you gain a stack, it refreshes the duration to thirty seconds again. When you hit 5 stacks, you should have 30 more seconds of the buff. However this is not the case, it is basing its duration off the original 30 second duration of when you first cast it, not the thirty seconds from getting the fifth stack.

**Comment by veremi** [comment]
Maybe small patch notes
/ Patch 2.0.1 (05-Dec-2006): Added. For 20 Rage, this instant ability adds an effect to your Warrior that increases Attack Power by 30 (50 at Max Rank) for 30 seconds. With each successful melee attack, this AP increase improves, until the effect has stacked 5 times (for a 150 AP Inc (250 Max Rank)). Rampage can only be triggered after delivering a Critical Hit.

How it should work ?
After activation, it will stack up to 5 times, refreshing itself with each stack. After 5 stacks, it stops refreshing and you have to refresh it manually.
How it work on Archangel ? 
Every crit refreshing ur rampage buff duration, but after 30sec rampage expires, refreshing rampage giving u 0 stacks and stack up after every crit 

Mb one more video from retail https://www.youtube.com/watch?v=zRH-GSaRtMg
Regards Wytoczka

**Comment by Applescruffs** [comment]
Already reported. 
#3639

**Comment by devidao** [comment]
it was fixed, but still not applied

**Comment by Yuskia** [comment]
So does that mean that a fix will be coming soon to a hotfix?

**Comment by Kaiitothewai** [comment]
It would appear it's still a issue, according to a friend of mine
